### PR TITLE
Snapshot schedule datasource[Powerscale 1.1]

### PR DESCRIPTION
**Pull request by nikitajoshi1**
# Description
  - Added Powerscale Snapshot Schedule Datasource and related tests

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Feature Pull Request

##### RESOURCE OR DATASOURCE NAME
Snapshot Schedule

##### OUTPUT
testSS
sscoverage
New  test for other filters:
otherFilterTest

##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by nikitajoshi1** [comment]
run e2e test

**Comment by nikitajoshi1** [comment]
run e2e test

File: examples/data-sources/powerscale_snapshot_schedule/datasource.tf [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+*/
+
+# PowerScale SMB shares provide clients network access to file system resources on the cluster
+
+# Returns a list of all the PowerScale Snapshot schedules
+data "powerscale_snapshot_schedule" "example_snapshot_schedule_all" {
+  filter {
+
+    # Used for query parameter, supported by PowerScale Platform API
+    # dir = "ASC"
```
Comment (doriac11) [inline examples/data-sources/powerscale_snapshot_schedule/datasource.tf:25]: Can you add a description to these and what they are filtering on?

**Review by doriac11 — APPROVED** [review]

**Comment by doriac11** [comment]
Can you add the coverage amount to the ticket?

**Comment by nikitajoshi1** [comment]
> Can you add the coverage amount to the ticket?

@doriac11 added coverage.

File: examples/data-sources/powerscale_snapshot_schedule/datasource.tf (hunk 1 above)
Comment (nikitajoshi1) [reply examples/data-sources/powerscale_snapshot_schedule/datasource.tf:26]: @doriac11 Updated with description.

**Comment by nikitajoshi1** [comment]
run e2e test

File: examples/data-sources/powerscale_snapshot_schedule/datasource.tf [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+# Returns a list of PowerScale Snapshot schedules based on path filter block
+data "powerscale_snapshot_schedule" "example_snapshot_schedule" {
+  filter {
+    # Used to specify path of snapshot schedule
+    path = "/ifs/DirNew1"
+
+    # Used for query parameter, supported by PowerScale Platform API
+    #The direction of the sort.Supported Values:ASC , DESC
```
Comment (doriac11) [inline examples/data-sources/powerscale_snapshot_schedule/datasource.tf:46]: I am not sure if the terraform user will care about these filters. It seems more useful to a client like a UI who would want sorting from the returned API. You might want to just create a filter on schedule name for the terraform user.

Comment (nikitajoshi1) [reply examples/data-sources/powerscale_snapshot_schedule/datasource.tf:46]: Added names as filter.

**Comment by nikitajoshi1** [comment]
run e2e test

**Comment by nikitajoshi1** [comment]
run e2e test

**Comment by nikitajoshi1** [comment]
run e2e test

**Comment by nikitajoshi1** [comment]
run e2e test

File: powerscale/provider/snapshot_schedule_datasource_test.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+				Config:      ProviderConfig + SnapshotScheduleDataSourceConfig,
+				ExpectError: regexp.MustCompile(`.*mock error*.`),
+			},
+		},
+	})
+}
+
+var SnapshotScheduleDataSourceConfig = `
+data "powerscale_snapshot_schedule" "test" {
+  filter {
```
Comment (shekhar-j) [inline powerscale/provider/snapshot_schedule_datasource_test.go:113]: @nikitajoshi1 can you try with other parameters also like dir, sort etc.?

File: powerscale/provider/snapshot_schedule_datasource.go [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+	tflog.Info(ctx, "Reading Snapshot Schedule data source ")
+	var ssPlan models.SnapshotScheduleDataSourceModel
+	var ssState models.SnapshotScheduleDataSourceModel
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &ssPlan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	totalSnapshotSchedules, err := helper.ListSnapshotSchedules(ctx, d.client, ssPlan.SnapshotScheduleFilter)
```
[the remark this answers is not in the document]
Comment (nikitajoshi1) [reply powerscale/provider/snapshot_schedule_datasource.go]: Updated to snapshotSchedules.

File: powerscale/provider/snapshot_schedule_datasource_test.go (hunk 3 above)
Comment (nikitajoshi1) [reply powerscale/provider/snapshot_schedule_datasource_test.go:131]: Added test for other filters.

**Review by shekhar-j — APPROVED** [review]

**Pull request merged**